partial_open_images_v7/partial_open_images_v7_dataset_builder.py

```diff
@@ -0,0 +1,454 @@
+                            "label": boxable_class_label,
+                            "confidence": np.int32,
+                            "bbox": tfds.features.BBoxFeature(),
+                            "is_occluded": np.int8,
+                            "is_truncated": np.int8,
+                            "is_group_of": np.int8,
+                            "is_depiction": np.int8,
+                            "is_inside": np.int8,
+                            "gender_presentation": gender_presentation_class_label,
+                            "age_presentation": age_presentation_class_label,
+                        }
+                    ),
+                }
+            ),
+            # If there's a common (input, target) tuple from the
+            # features, specify them here. They'll be used if
+            # `as_supervised=True` in `builder.as_dataset`.
+            supervised_keys=None,  # Set to `None` to disable
+            homepage="https://storage.googleapis.com/openimages/web/index.html",
+        )
+
+    def _split_generators(self, dl_manager: tfds.download.DownloadManager):
+        """Returns SplitGenerators."""
+
+        paths = dl_manager.download_and_extract(_URLS)
+
+        url_to_image_id = _load_image_ids(paths["image_ids"])
+
+        return {
+            "train": self._generate_examples(
+                dl_manager.manual_dir / "train", "train", paths, url_to_image_id
+            ),
+            "validation": self._generate_examples(
+                dl_manager.manual_dir / "validation",
+                "validation",
+                paths,
+                url_to_image_id,
+            ),
+            "test": self._generate_examples(
+                dl_manager.manual_dir / "test", "test", paths, url_to_image_id
+            ),
+        }
+
+    def _generate_examples(self, path, split, paths, url_to_image_id):
+        """Yields examples."""
+
+        def load(names):
+            csv_positions = [0] * len(names)
+            return functools.partial(
+                _load_objects, [paths[name] for name in names], csv_positions
+            )
+
+        def load_boxes(name):
+            csv_positions = [0]
+            return functools.partial(_load_bboxes, paths[name], csv_positions)
+
+        objects = load([f"{split}_human_labels", f"{split}_machine_labels"])
+
+        bboxes = load_boxes(f"{split}-annotations-bbox")
+
+        objects = objects(None)
+        bboxes = bboxes(None)
+        if split != "train":
+            miaps = _load_miaps(paths[f"{split}-annotations-miap"])
+
+        for f in path.glob("*.jpg"):
+            file_name = os.path.basename(f)
+            image_id = int(os.path.splitext(url_to_image_id[file_name])[0], 16)
+            image_objects = [obj._asdict() for obj in objects.get(image_id, [])]
+            image_objects = [
+                image_object
+                for image_object in image_objects
+                if image_object["confidence"] >= 5
```

Comment:
Turn into configuration

Reply:
Maybe it doesn't make so much sense here, but rather after loading the dataset. Should we include labels even with lower confidence than 5 in the full open images version 7 dataset.